cirq-core/cirq/experiments/t1_decay_experiment_test.py

```diff
@@ -165,6 +167,72 @@ def test_all_off_results():
     )
 
 
+def test_constant():
+    result_100 = cirq.experiments.T1DecayResult(
+        data=pd.DataFrame(
+            columns=['delay_ns', 'false_count', 'true_count'],
+            index=range(4),
+            data=[
+                [100.0, 6, 4],
+                [400.0, 10, 0],
+                [700.0, 10, 0],
+                [1000.0, 10, 0],
+            ],
+        )
+    )
+    assert np.isclose(result_100.constant, 100, 5)
+
+    result_400 = cirq.experiments.T1DecayResult(
+        data=pd.DataFrame(
+            columns=['delay_ns', 'false_count', 'true_count'],
+            index=range(4),
+            data=[
+                [100.0, 0, 10],
+                [400.0, 6, 4],
+                [700.0, 10, 0],
+                [1000.0, 10, 0],
+            ],
+        )
+    )
+    assert np.isclose(result_400.constant, 400, 5)
+
```

Comment:
Could we add one more test here that explicitly generates an exponential decay of some kind:

```python3
class GradualDecay(cirq.NoiseModel):
    def noisy_moment(self, moment, system_qubits):
        duration = max(
            (
                op.gate.duration
                for op in moment.operations
                if isinstance(op.gate, cirq.WaitGate)
            ),
            default=cirq.Duration(nanos=0),
        )
        if duration >cirq.Duration(nanos=0):
            # Found a wait gate in this moment.
            return cirq.amplitude_damp(duration.total_nanos() / 1000).on_each(system_qubits)
        return moment
```
And then verifies that the constant returned by the `constant` method is the one that you used for the setup ?

Reply:
I agree that testing it on an explicit exponential decay is a good idea; however, the t1_decay_experiment method behaves differently with a simulator than it does with hardware (from what I understand, there is only one wait gate added to the circuit and so the simulated noise is not applied properly "over time" since it is only applied after the one gate). All of this to say, I think generating a T1DecayResult object with explicitly exponential data might be a better way to go for this test instead of actually running the T1 decay experiment on a noisy simulator. What do you think?

Reply:
Hey @MichaelBroughton - for some clarification, there is a follow-up PR coming after this which will add an option to the T1 decay experiment to support simulators using arbitrary noise models. That PR seems like the logical place for the test you describe, since it depends on the simulator/noise (unlike the current PR, which is agnostic to the source of the data). What do you think?

Reply:
Sure thing. We can leave out a more complicated test for now and place it elsewhere in this future PR.